### PR TITLE
Boost simd

### PR DESCRIPTION
**Pull request by johnmcfarlane**

### CODE REVIEW

File: src/test/boost.simd.cpp [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    using sg14::fixed_point;
+    using boost::simd::pack;
+
+    template<class LhsT, std::size_t LhsN, int LhsExponent, class RhsT, std::size_t RhsN, int RhsExponent>
+    void assert_identical(const fixed_point <pack<LhsT, LhsN>, LhsExponent> &lhs,
+                          const fixed_point <pack<RhsT, RhsN>, RhsExponent> &rhs) {
+        static_assert(LhsN == RhsN, "pack is the wrong size");
+        static_assert(std::is_same<LhsT, RhsT>::value, "pack is the wrong type");
+        auto equal = lhs == rhs;
+        if (!boost::simd::all(equal)) {
```
Comment (jfalcou) [inline src/test/boost.simd.cpp]: this is maybe a job for boost::simd::compare_equal

File: include/sg14/auxiliary/boost.simd.h [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+#include <sg14/fixed_point>
+
+#include <boost/simd/pack.hpp>
+
+namespace sg14 {
+    template<class Rep, std::size_t N, int Exponent>
+    ::std::ostream& operator<<(::std::ostream& out, const fixed_point<boost::simd::pack<Rep, N>, Exponent>& fp)
+    {
+        return out << static_cast<boost::simd::pack<long double, N>>(fp);
```
Comment (jfalcou) [inline include/sg14/auxiliary/boost.simd.h:21]: is it actually needed or can fp just returns its internal pack via some public API ?

Comment (johnmcfarlane) [reply include/sg14/auxiliary/boost.simd.h:21]: This is a hack for debugging mostly. You can get the `rep_` member with the `data()` method.

File: src/test/boost.simd.cpp (hunk 1 above)
Comment (johnmcfarlane) [reply src/test/boost.simd.cpp]: Will change this. Although in the same spirit of using `explicit` on conversion operators, have you considered an `explicit operator bool()`?

**Comment by johnmcfarlane** [comment]
Thanks for the feedback, @jfalcou. Note that due to lack of support in fixed_point's CI, Boost.SIMD integration is only tested on OS X.

**Pull request merged**